### PR TITLE
Implements new UI for dockup

### PR DESCRIPTION
**Pull request by aksh93**
- Root page
image
- Configure
image

### CODE REVIEW

File: apps/dockup_ui/lib/dockup_ui/endpoint.ex [hunk 1]
```diff
@@ -9,7 +9,7 @@ defmodule DockupUi.Endpoint do
   # when deploying your static files in production.
   plug Plug.Static,
     at: "/", from: :dockup_ui, gzip: false,
-    only: ~w(css fonts images js favicon.ico robots.txt)
+    only: ~w(css fonts images js favicon.ico robots.txt icons)
```
Comment (emilsoman) [inline apps/dockup_ui/lib/dockup_ui/endpoint.ex]: Do we need font-awesome now that we have an icons directory?

File: apps/dockup_ui/package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "brunch": "^2",
     "copycat-brunch": "~1.1.0",
     "jquery": "3.1.1",
+    "npm": "^6.1.0",
```
Comment (emilsoman) [inline apps/dockup_ui/package.json]: Why is npm added?

File: apps/dockup_ui/web/static/css/app.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  margin-top: 5px;
+
+  li {
+    width: 100%;
+    padding: 5px;
+    border: 1px solid #3E83FF;
+    z-index: 1000;
+    opacity: 1;
+  }
+}
```
Comment (emilsoman) [inline apps/dockup_ui/web/static/css/app.scss]: what is "rbt" ?

File: apps/dockup_ui/web/static/js/components/deployment_form.js [hunk 4]
```diff
@@ -36,7 +36,7 @@ class DeploymentForm extends Component {
 
   handleDeployClick(e) {
     this.setState({deployment: null});
-    e.preventDefault();
+    // e.preventDefault();
```
Comment (emilsoman) [inline apps/dockup_ui/web/static/js/components/deployment_form.js]: Why was this commented?

File: apps/dockup_ui/web/static/js/components/deployment_status.js [hunk 5]
```diff
@@ -1,60 +1,32 @@
...
+  "starting": "icon-sync",
+  "checking_urls": "icon-sync",
+  "started": "icon-deployed",
+  "hibernating": "icon-sync",
+  "hibernated": "icon-hibernated",
+  "waking_up": "icon-sync",
+  "restarting": "icon-sync",
+  "deleting": "icon-sync",
+  "deleted": "icon-deleted",
+  "failed": "icon-errored"
```
Comment (emilsoman) [inline apps/dockup_ui/web/static/js/components/deployment_status.js]: These statuses are old. Valid states are:

```elixir
queued, starting, waiting_for_urls, started, hibernating, hibernated, waking_up, deleting, deleted, failed, restarting
```
ie, we removed processing and cloning_repo, also renamed "checking_urls" to "waiting_for_urls"

File: apps/dockup_ui/web/static/js/components/deployment_status.js [hunk 6]
```diff
@@ -1,60 +1,32 @@
...
 }
 
 const DeploymentStatus = ({status}) => {
-  let textClass = `text-${getStatusColorClass(status)}`;
+  console.log(status);
```
Comment (emilsoman) [inline apps/dockup_ui/web/static/js/components/deployment_status.js]: Stray console.log here.

File: apps/dockup_ui/lib/dockup_ui/endpoint.ex (hunk 1 above)
Comment (iffyuva) [reply apps/dockup_ui/lib/dockup_ui/endpoint.ex]: Removed font-awesome from `package.json`

File: apps/dockup_ui/package.json (hunk 2 above)
Comment (iffyuva) [reply apps/dockup_ui/package.json]: Removed it.

File: apps/dockup_ui/web/static/css/app.scss (hunk 3 above)
Comment (iffyuva) [reply apps/dockup_ui/web/static/css/app.scss]: These classes are related to `react-bootstrap-typeahead`

File: apps/dockup_ui/web/static/js/components/deployment_form.js (hunk 4 above)
Comment (iffyuva) [reply apps/dockup_ui/web/static/js/components/deployment_form.js]: Reverted this change.

File: apps/dockup_ui/web/static/js/components/deployment_status.js (hunk 5 above)
Comment (iffyuva) [reply apps/dockup_ui/web/static/js/components/deployment_status.js]: Updated these statuses, please check.

File: apps/dockup_ui/web/static/js/components/deployment_status.js (hunk 6 above)
Comment (iffyuva) [reply apps/dockup_ui/web/static/js/components/deployment_status.js]: Remove this `console.log`

**Pull request merged**